bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/handler/HaperoBridgeHandler.java

```diff
@@ -0,0 +1,635 @@
+                        errorMsg = "@text/offline.monitoring.failed";
+                    }
+                } else {
+                    logger.warn("Failed to create watchservice");
+                    faultOcurred = true;
+                    errorMsg = "@text/offline.monitoring.failed";
+                }
+
+                /* Start the File Watcher Runnable if a Watcher could be created */
+                if (!faultOcurred) {
+                    fileWatcherJob = scheduler.schedule(fileWatcherTask, 0, TimeUnit.SECONDS);
+                }
+            }
+
+            if (!faultOcurred) {
+                /*
+                 * If a watcher could be created, start a discovery.
+                 * The Thing status will be set to ONLINE after the first successful update
+                 */
+                updateStatus(ThingStatus.UNKNOWN);
+
+                scheduler.execute(() -> {
+                    updateThings();
+
+                    if (discoveryService != null) {
+                        discoveryService.startScan();
+                    }
+                });
+                logger.trace("Bridge initialized");
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+            }
+        } else {
+            /* A Watcher could not be created, set Thing to OFFLINE. Manual restart is required. */
+            cancelAllJobs();
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            logger.trace("Bridge switched to offline mode because of configuration error");
+        }
+    }
+
+    /**
+     * Get a list of of all devices reported from the control system
+     *
+     * @return String List of all devices (may be empty)
+     */
+    public String[] getAllDevices() {
+        return deviceList.keySet().toArray(String[]::new);
+    }
+
+    /**
+     * Get the {@link Device} corresponding to the given deviceID
+     *
+     * @param deviceID ID for which to get the Device
+     * @return Device or null if the Device could not be found
+     */
+    public @Nullable Device getDevice(String deviceID) {
+        Device result = deviceList.get(deviceID);
+        if (result == null) {
+            logger.warn("getDevice: No device found for ID {}.", deviceID);
+        }
+        return result;
+    }
+
+    /**
+     * Set the discoveryService
+     *
+     * @param discoveryService new discoveryService
+     */
+    public void setDiscoveryService(HaperoDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    /**
+     * Update the device data from an InputStream (for FTP access)
+     *
+     * @param stream InputStream containing the raw device data.
+     */
+    private void updateDeviceList(InputStream stream) {
+        String result;
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        try {
```

Comment:
BufferedReader is Closeable, you could use try with resource syntax : 
`try (BufferedReader reader = new Buff....) {`

Reply:
Cool, i didn't know you could do that. I'll change it.